rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -173,13 +229,62 @@ fn base_connection_from_dbus(conn: &OwnedNestedHash) -> Option<BaseConnection> {
         ..Default::default()
     };
 
+    if let Some(interface) = connection.get("interface-name") {
+        let interface: &str = interface.downcast_ref()?;
+        base_connection.interface = interface.parse().unwrap();
+    }
+
+    if let Some(match_config) = conn.get("match") {
+        base_connection.match_config = match_config_from_dbus(match_config)?;
+    }
+
     if let Some(ipv4) = conn.get("ipv4") {
         base_connection.ipv4 = ipv4_config_from_dbus(ipv4)?;
     }
 
     Some(base_connection)
 }
 
+fn match_config_from_dbus(
+    match_config: &HashMap<String, zvariant::OwnedValue>,
+) -> Option<MatchConfig> {
+    let mut match_conf = MatchConfig::default();
+
+    if let Some(drivers) = match_config.get("driver") {
+        let drivers = drivers.downcast_ref::<zbus::zvariant::Array>()?;
+        for driver in drivers.get() {
+            let driver: &str = driver.downcast_ref()?;
+            match_conf.driver.push(driver.parse().unwrap());
```

Comment:
What are you "parsing"? Just converting to a String should be enough.

```suggestion
            match_conf.driver.push(driver.to_string());
```